model-inference/decisionTree/experiments/model_helper.py

```diff
@@ -7,16 +8,27 @@
 from sklearn.metrics import classification_report, mean_squared_error
 import treelite_runtime
 
+dataset_folder = "dataset/"
+
 def calculate_time(start_time,end_time):
     diff = (end_time-start_time)*1000
     return diff
 
-def load_data_from_pickle(dataset):
-    raise NotImplementedError
+def load_data_from_pickle(dataset, config, suffix, time_consume):
+    start_time = time.time()
+    pkl_path = relative2abspath(dataset_folder, f"{config[dataset]['filename']}_{suffix}.pkl")
+    dataframe = pickle.load(open(pkl_path, "rb"))
+    end_time = time.time()
+    data_loading_time = calculate_time(start_time,end_time)
+    if time_consume is not None:
+        time_consume["data loading time"] = data_loading_time
+    print(f"Time Taken to load {dataset} as a dataframe is: {data_loading_time}")
+    # import pdb; pdb.set_trace()
+    return dataframe
 
 def fetch_data(dataset, config, suffix, time_consume=None):
-    if dataset == "covtype":
-        load_data_from_pickle(dataset)
+    if dataset == "epsilon": # Does not fit PostgreSQL
```

Comment:
If I understand this correctly, since it does not fit in Postgres, we read/fetch the data directly from .pkl file. If yes, this approach is not consistent with other datasets, right? 

Considering that, are we still going ahead with the experiments for this dataset? Or, keeping epsilon aside for the moment?

Reply:
Yes, this approach is inconsistent with other datasets, but only affects the data loading time. However, we can still measure the inference time.